File: ui-tester/src/main/kotlin/uitester/vm/WifiScanNVM.kt
package uitester.vm

import NextionRenderer
import RendereableNVM

class WifiScanNVM(renderer: NextionRenderer) : RendereableNVM(renderer) {
    override fun checkMatch(data: ByteArray): Boolean {
        return data.size == 2 && data[0] == 0x00.toByte() && data[1] == 0x00.toByte()
    }

    override fun control(data:ByteArray) {
        renderer.render("wifiSsid.scanResult.txt=\"network1;network2;network3\"")
        renderer.render("page wifiSsid")
    }
}

File: micropython/src/nvm/wifi_scan_nvm.py
from nextion import *
from ae import *

import network

class WiFiScanNVM(NextionViewModel):
    def __init__(self, renderer: NextionRenderer, wlan: network.WLAN):
        super().__init__(renderer)
        self.wlan = wlan

    def checkMatch(self, data: bytearray):
        return len(data) == 2 and data[0] == RequestType_Data and data[1] == EntityType_SSIDs

    def control(self, data: bytearray):
        scans = self.wlan.scan()
        self.renderer.render('wifiSsid.scanResult.txt="', insertDelimeter = False)
        first = True
        for scan in scans:
            if not first:
                self.renderer.render(';', insertDelimeter = False)
            
            self.renderer.render(scan[0], insertDelimeter = False)
            first = False

        self.renderer.render('"')
        self.renderer.render('page wifiSsid')
